src/soc.c

```diff
@@ -1,11 +1,46 @@
 #include "eeprom_config.h"
 #include "board.h"
+#include "state_types.h"
 
-void SOC_Init(void) {
-    // EEPROM_WriteCCPage_Num(0,18);
+static uint32_t soc;
+static uint32_t last_tick_soc;
+
+
+#define MAX_CHARGE 10000000
+
+void SOC_Init(/*fullycharged?,*/ uint32_t ms_ticks) {
+	// if(/*fully_charged*/){
+	// 	init_soc = MAX_CHARGE;
+	// 	soc = init_soc;
+	// 	/*fully_charged = false;*/
+	// 	last_tick_soc = ms_ticks;
+	// } else {
+		last_tick_soc = ms_ticks;
+		soc = EEPROM_LoadCCPage_Num(0);
+		
+	//}
 }
 
-uint32_t SOC_Estimate(void) {
-    // return EEPROM_LoadCCPage_Num(0);
-    return 0;
+#define DELTA_T 50
+uint32_t SOC_Estimate(BMS_INPUT_T* bms_input, uint32_t ms_ticks) {
+
+	//rectangular integration
+	if((ms_ticks - last_tick_soc)> DELTA_T){
+		last_tick_soc = ms_ticks;
+		//units are coulombs
+		soc = (soc - (bms_input->pack_status->pack_current_mA) * DELTA_T);
```

Comment:
Displayed SoC should ideally be in a 0-100 scale on the dash, so we have a human-interpretable value. Might be interested in calibrating a mapping.